packages/launchpad/src/global/GlobalPageHeader.vue

```diff
@@ -5,30 +5,59 @@
     class="min-w-200px w-80% flex-grow"
     :placeholder="t('globalPage.searchPlaceholder')"
   />
+  <input
+    ref="fileInputRef"
+    type="file"
+    class="hidden"
+    webkitdirectory
+    webkit-relative-path
+    @change="handleFileSelection"
+  >
   <Button
     :prefix-icon="IconPlus"
+    aria-controls="fileupload"
     prefix-icon-class="text-center justify-center text-lg"
     class="w-20% min-w-120px text-size-16px h-full focus-within:ring-2 focus-within:ring-offset-2 focus-within:ring-indigo-500"
-    @click="$emit('new-project')"
+    @click="handleButtonClick"
   >
     {{ t('globalPage.newProjectButton') }}
   </Button>
 </template>
 
 <script lang="ts" setup>
+import { ref } from 'vue'
 import Button from '@cy/components/Button.vue'
 import Input from '@cy/components/Input.vue'
 import IconPlus from '~icons/mdi/plus'
-import { useModelWrapper } from '@packages/frontend-shared/src/composables'
+import { useAddProject, useModelWrapper } from '@packages/frontend-shared/src/composables'
 import { useI18n } from '@cy/i18n'
 
+const fileInputRef = ref<HTMLInputElement>()
 const { t } = useI18n()
+const { addProject } = useAddProject()
 
 const props = defineProps<{
   modelValue: string
 }>()
 
-const emits = defineEmits(['update:modelValue', 'new-project'])
+function handleButtonClick () {
+  fileInputRef.value.click()
+}
+
+function handleFileSelection (e: Event) {
+  const target = e.target as HTMLInputElement
+  const dirPath = getDirectoryPath(target.files)
+
+  addProject(dirPath)
+}
```

Comment:
This line is really important, but it isn't covered by tests because it uses a side-effect (global state via gql) to affect the rest of the UI in the app.

To make this component more testable, I think

*   the `addProject` event should be emitted
*   the mutation should be called from the parent.
*   Since the parent is responsible for displaying the new project, I think those tests should be updated to validate that "when a project is uploaded, then...".